reconcile/dynatrace_token_provider/integration.py

```diff
@@ -231,13 +232,34 @@ def reconcile(self, dry_run: bool, dependencies: Dependencies) -> None:
                                 ocm_env_name=ocm_env_name,
                             )
                     except Exception as e:
-                        unhandled_exceptions.append(
-                            f"{ocm_env_name}/{cluster.organization_id}/{cluster.external_id}: {e}"
-                        )
+                        unhandled_exceptions.append((
+                            e,
+                            f"{ocm_env_name}/{cluster.organization_id}/{cluster.external_id}: {e}",
+                        ))
                 self._expose_token_metrics()
 
-        if unhandled_exceptions:
-            raise ReconcileErrorSummary(unhandled_exceptions)
+        unexpected_errors_cnt = 0
+        for err, _ in unhandled_exceptions:
+            if (
+                isinstance(err, OCMAPIError)
+                and err.status_code >= 500
+                and (
+                    "api.integration.openshift.com" in err.endpoint
+                    or "api.stage.openshift.com" in err.endpoint
+                )
+            ):
+                # OCMAPI in int/stage can fail. Lets for now ignore those
+                continue
+            unexpected_errors_cnt += 1
+
+        # Lets only fail integration on unexpected errors
+        error_messages = [msg for _, msg in unhandled_exceptions]
+        if unexpected_errors_cnt:
+            raise ReconcileErrorSummary(error_messages)
```

Comment:
since only boolean check needed, can use `any`

```python
unexpected_errors = any(not integration_or_stage_500(err) for err, _ in unhandled_exceptions)
```